lib/Authentication/Core/MerchantConfiguration.php

```diff
@@ -1441,6 +1751,135 @@ private function validateMLEConfiguration(){
                 throw $exception;
             }
         }
+
+        /*
+         * RESPONSE (OUTBOUND) MLE VALIDATION
+         * Trigger if global flag OR any per-API response flag is true.
+         */
+        $responseMleConfigured = (bool)$this->enableResponseMleGlobally;
+        foreach ($this->internalMapToControlResponseMLEonAPI as $flag) {
+            if ($flag) { $responseMleConfigured = true; break; }
+        }
+
+        if ($responseMleConfigured) {
+            if (strcasecmp($this->authenticationType, GlobalParameter::JWT) !== 0) {
+                $error_message = "Response MLE is only supported for JWT authentication type.";
+                $exception = new AuthException($error_message, 0);
+                self::$logger->error($error_message);
+                throw $exception;
+            }
+
+            // Validate responseMleKID
+            if (empty(trim($this->responseMleKID))) {
+                $error_message = "Response MLE enabled but responseMleKID is not set.";
+                $exception = new AuthException($error_message, 0);
+                self::$logger->error($error_message);
+                throw $exception;
+            }
+
+            $hasFilePath = !empty($this->responseMlePrivateKeyFilePath);
+            $hasInMemoryKey = !empty($this->responseMlePrivateKey);
+
+            if (!$hasFilePath && !$hasInMemoryKey) {
+                $error_message = "Response MLE enabled but neither responseMlePrivateKeyFilePath nor responseMlePrivateKey provided. Provide exactly one.";
+                $exception = new AuthException($error_message, 0);
+                self::$logger->error($error_message);
+                throw $exception;
+            }
+
+            if ($hasFilePath && $hasInMemoryKey) {
+                $error_message = "Both responseMlePrivateKeyFilePath and responseMlePrivateKey supplied. Provide only one.";
+                $exception = new AuthException($error_message, 0);
+                self::$logger->error($error_message);
+                throw $exception;
+            }
+
+            if ($hasFilePath) {
+                if (
+                    !file_exists($this->responseMlePrivateKeyFilePath) ||
+                    !is_readable($this->responseMlePrivateKeyFilePath) ||
+                    !is_file($this->responseMlePrivateKeyFilePath)
+                ) {
+                    $error_message = "Response MLE private key file not found or not readable at " . $this->responseMlePrivateKeyFilePath;
+                    $exception = new AuthException($error_message, 0);
+                    self::$logger->error($error_message);
+                    throw $exception;
+                }
+            } else {
+                // Basic sanity check for in-memory key
+                if (!is_object($this->responseMlePrivateKey)) {
+                    $error_message = "Response MLE private key object is invalid. Must be OpenSSLAsymmetricKey";
```

Comment:
This error message is misleading. You are checking if it is an object. What does that have to do with OpenSSL?